src/github/client/key-locator.ts

```diff
@@ -0,0 +1,59 @@
+import { GitHubServerApp } from "~/src/models/github-server-app";
+import { resolve } from "path";
+import { existsSync, readFileSync } from "fs";
+import { envVars } from "~/src/config/env";
+import isBase64 from "is-base64";
+
+const begin = "-----BEGIN RSA PRIVATE KEY-----";
+const end = "-----END RSA PRIVATE KEY-----";
+/**
+ * Look for a Github app's private key
+ */
+export const keyLocator = async (gitHubAppId?: number) => {
+	if (gitHubAppId) {
+		const gitHubServerApp = await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
+		if (gitHubServerApp) {
+			return await gitHubServerApp.decrypt("privateKey");
+		}
+	}
+	else {
+		// Fetch private key for github cloud
+		if (envVars.PRIVATE_KEY) {
+			let privateKey = envVars.PRIVATE_KEY;
+			if (isBase64(privateKey)) {
+				privateKey = Buffer.from(privateKey, "base64").toString();
+			}
+
+			if (privateKey.includes(begin) && privateKey.includes(end)) {
+				// newlines are escaped
+				if (privateKey.indexOf("\\n") !== -1) {
+					privateKey = privateKey.replace(/\\n/g, "\n");
+				}
+				// newlines are missing
+				if (privateKey.indexOf("\n") === -1) {
+					privateKey = addNewlines(privateKey);
```

Comment:
can you explain this further?

Reply:
In case private key wrapped in a single line, this will correct the format as -
-----BEGIN RSA PRIVATE KEY-----
aaaaaaaaaaaa
bbbbbbbbb
-----END RSA PRIVATE KEY-----

This case also handled in Probot. I'm not sure whether we need this one or not.

Reply:
yeah, we really shouldn't be doing that since it means the key has been escaped.  I'd rather not fix PIBKAC issues ;)